workflow/src/main/java/com/synopsys/integration/alert/workflow/message/NotificationReceiver.java

```diff
@@ -82,10 +88,14 @@ public void handleEvent(NotificationReceivedEvent event) {
             logger.trace("Processing Page: {}. New pages found: {}",
                 numPagesProcessed,
                 pageOfAlertNotificationModels.getTotalPages());
+            hasNotTimedOut = DateUtils.createCurrentDateTimestamp().isBefore(timeLimit);
         }
         if (numPagesProcessed == MAX_NUMBER_PAGES_PROCESSED) {
             logger.warn("Receiver reached upper page limit of pages processed: {}, exiting.", MAX_NUMBER_PAGES_PROCESSED);
         }
+        if (!hasNotTimedOut) {
+            logger.warn("Receiver has timed out after 2 hours.");
```

Comment:
Is there any way to make this dynamic or is that not important?

Reply:
Right now we are arbitrarily adding 2 hours in the timeLimit, so I think this should be okay unless someone else has an issue with it and I can save that value to a variable. This should hopefully be resolved in the changes in 6.5.0 and will no longer need this timeout. If anything I can put in an extra comment here that this should be removed once those changes are made.

Reply:
This will hopefully become unnecessary in 6.5.0 and removed.